tests/cxx/no_implicit_typedef_reporting.cc

```diff
@@ -0,0 +1,34 @@
+//===--- no_implicit_typedef_reporting.cc - test input file for iwyu ------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU_ARGS: -I . \
+//            -Xiwyu --check_also="tests/cxx/no_implicit_typedef_reporting-d2.h"
+
+// Tests that IWYU doesn't suggest to include typedef-containing header due to
+// implicit-only use of the typedef, i.e. when it is not explicitly written
+// in a source.
+
+#include "tests/cxx/no_implicit_typedef_reporting-d1.h"
+#include "tests/cxx/no_implicit_typedef_reporting-d2.h"
+
+void Fn() {
+  (void)sizeof(Struct::typedefed);
```

Comment:
I played with some variations:
```
template<class T>
struct Template {
  T t;
  Struct s;
};

void Fn() {
  // Compile-time expression.
  (void)sizeof(Struct::typedefed);

  // Member expression.
  Struct s;
  int i = s.typedefed + 10;

  // Pointer arithmetic.
  void* p = &s.typedefed + 10;

  // Template, both argument and explicitly used.
  Template<Struct> temp;
  (void)temp.t.typedefed;
  (void)temp.s.typedefed;
}
```

At least the pointer arithmetic example reports a use of `Int` on master, so that could at least be worth covering.

The others seem like good standard examples even if no uses are triggered on master either.

Reply:
I've applied your suggestion with some changes. Firstly, it should be noted that a use of a member variable after the dot in a member expression doesn't constitute the full use of its type. I could convert `typedefed` underlying type into a `struct` and name its field (`s.typedefed.field`) to test a use in the member expression, if you insist.
Secondly, binary operators are either builtin operators, as in this case, or user-defined operators which may be declared somewhere else, not necessarily in the class or nearby it. So, a use of such operators doesn't generally constitute the full type use.
In my variant, all the three lines are triggered on master.

Reply:
No, I think these additional examples are perfectly fine, thanks!